File: app/src/main/java/xxx/RequestHeaderSetter.java
package xxx;

import javax.annotation.Nullable;

import io.opentelemetry.context.propagation.TextMapSetter;
import okhttp3.Request;

enum RequestHeaderSetter implements TextMapSetter<Request.Builder> {
    INSTANCE;

    @Override
    public void set(@Nullable Request.Builder carrier, String key, String value) {
        if (carrier == null) {
            return;
        }
        carrier.header(key, value);
    }
}

File: app/src/main/java/network/RequestHeaderSetter.kt
package network

import io.opentelemetry.context.propagation.TextMapSetter
import okhttp3.Request.Builder
object RequestHeaderSetter : TextMapSetter<Builder> {
    override fun set(carrier: Builder?, key: String, value: String) {
        carrier?.header(key, value)
    }
}
